motionplan/collision.go

```diff
@@ -319,3 +335,33 @@ func createUniqueCollisionMap(geoms []spatial.Geometry) (map[string]spatial.Geom
 	}
 	return geomMap, nil
 }
+
+func firstMovingParentOrself(fs *referenceframe.FrameSystem, f referenceframe.Frame) referenceframe.Frame {
+	var err error
+	for f != fs.World() {
+		if len(f.DoF()) > 0 {
+			return f
+		}
+
+		f, err = fs.Parent(f)
+		if err != nil {
+			panic(err) // should be impossible
+		}
+	}
+	return f
+}
+
+func skipCollisionCheck(fs *referenceframe.FrameSystem, xName, yName string) bool {
```

Comment:
I think I was able to break the correctness of this function. Where I can craft a case such that a hypothetical left and right arm won't get collision checked. But I went through some mental gymnastics to do so. I don't think it's worth elaborating on here though. Unless you were specifically concerned with this part of the code and assumptions it makes re: the user's frame configuration.